### PR TITLE
Bug/18731 features tests for help link

### PR DESCRIPTION
**Pull request by rmagnum2002**
https://community.openproject.org/work_packages/18731

### CODE REVIEW

File: features/step_definitions/help_relinking.rb [hunk 1]
```diff
@@ -17,15 +17,21 @@
...
+Given(/^the help link setting points to help page$/) do
   new_values = Setting.plugin_openproject_help_link
-  new_values['help_link_target'] = "/projects/#{p.identifier}"
+  new_values["help_link_target"] = OpenProject::Info.help_url
   Setting.plugin_openproject_help_link = new_values
 
   Redmine::MenuManager.map :top_menu do |menu|
     menu.delete :help
-    menu.push :help, OpenProject::Info.help_url, last: true
+    menu.push :help, OpenProject::Info.help_url, :last => true
```
Comment (linki) [inline features/step_definitions/help_relinking.rb]: use the 1.9 syntax here

File: features/step_definitions/help_relinking.rb [hunk 2]
```diff
@@ -17,15 +17,21 @@
...
 #++
 
-Given /^the help link setting points to the project (.+)$/ do |project_name|
-  project_name.gsub!("\"", '')
-  p = Project.find_by_name(project_name)
+Given(/^the help link setting points to help page$/) do
   new_values = Setting.plugin_openproject_help_link
-  new_values['help_link_target'] = "/projects/#{p.identifier}"
+  new_values["help_link_target"] = OpenProject::Info.help_url
   Setting.plugin_openproject_help_link = new_values
```
Comment (linki) [inline features/step_definitions/help_relinking.rb]: that's a no-op.

`new_values["help_link_target"] = OpenProject::Info.help_url` expands to
`new_values["help_link_target"] = Setting.plugin_openproject_help_link['help_link_target']` so this line becomes

```
Setting.plugin_openproject_help_link = Setting.plugin_openproject_help_link`
```

 try to use more literal values when testing, like it was before. changing the the `default` help link target to `foo` will still pass your tests.

Comment (linki) [inline features/step_definitions/help_relinking.rb]: another good way is to make your test fail before you make it pass. in this case: configure a different url and see that it doesn't go to the help page (as the scenario title suggests)

File: features/link_help_to_project.feature [hunk 3]
```diff
@@ -22,18 +22,7 @@ Feature: Link help to feature
...
-      | Name | loginproject |
-    And there is a role "Loginrole"
-    And the user "bob" is a "Loginrole" in the project "loginproject"
-    And I am logged in as "bob"
-    When I follow "Help" within "#top-menu"
-    Then I should be on the page of the project "Support"
+  Scenario: if help link settings not changed it will point to /help
+    Given the help link setting points to help page
+    When I visit the root page
+    Then I should have the help link in the top menu
```
Comment (linki) [inline features/link_help_to_project.feature:28]: thanks for removing the useless boilerplate steps @rmagnum2002 :+1:

**Comment by linki** [comment]
@rmagnum2002 can you please check if the tests pass without this PR?

on second look i think this might not be needed. the test for the old functionality is: "the help link can be defined per project via the settings in the database". this is desired and doesn't need to be changed.

the spec that we **need** to add from the related PR https://github.com/finnlabs/openproject-help_link/pull/9 is "spec that the default is https://www.openproject.org/help".

i can not merge this like this.

**Comment by rmagnum2002** [comment]
@linki  It shouldn't pass, as I remember it was looking for a link that has url as `/projects/support`, and it tries to find a project named support that doesn't exists. But I'll run them again and get back with a feedback

**Comment by linki** [comment]
@rmagnum2002 right, it fails because the thing behind the link doesn't exist. it does **not** fail because of wrong functionality. maybe you can just change the very specific expectation "Then I should be on the page of the project "Support"" to "Then help link should point to /projects/help" or similar.

this way you don't have to change anything. we just need to make sure that the value in there is correct (not an invalid url).

still, a spec corresponding to your last PR (default is `foo`) needs to be added.

**Comment by oliverguenther** [comment]
Closing due to deprecation

**Pull request closed without merging**